modules/Microsoft.Compute/virtualMachines/deploy.bicep

```
@@ -2,6 +2,9 @@
 @description('Optional. The name of the virtual machine to be created. You should use a unique prefix to reduce name collisions in Active Directory. If no value is provided, a 10 character long unique string will be generated based on the Resource Group\'s name.')
 param name string = take(toLower(uniqueString(resourceGroup().name)), 10)
 
+@description('Optional. Can be used if the computer name needs to be different from the Azure VM resource name. If not used, the resource name will be used as computer name.')
+param vmCustomComputerName string = ''
```

I'd suggest to keep the parameter name aligned with the API. Also, to directly assign the `name` value as default value instead of an empty value.

```suggestion
param computerName string = name
```